NuRadioReco/utilities/diodeSimulator.py

```diff
@@ -150,20 +153,29 @@ def calculate_noise_parameters(self,
         power_std: float
             Standard deviation of the diode-filtered noise
         """
-        noise = NuRadioReco.framework.channel.Channel(0)
 
-        long_noise = channelGenericNoiseAdder().bandlimited_noise(min_freq=min_freq,
-                                        max_freq=max_freq,
-                                        n_samples=10000,
-                                        sampling_rate=sampling_rate,
-                                        amplitude=amplitude,
-                                        type=type)
+        power_mean_list = []
+        power_std_list = []
+
+        for i_sample in range(n_samples):
```

Comment:
Is this just a handy coincidence that you do the calcualtion as many times as you have samples in the trace or am I missing the logic here?

Reply:
Otherwise, I would suggest to add another parameter. The documentation is also a bit misleading now, as n_samples is both the loop and the number of samples, as the name suggests.

Reply:
They are different parameters. I'm going to change the name. Another option would be to increase the trace by a factor of 10000. I'm not sure which method would be faster, but in any case this has to be done only once and the loop seems just fine.

Reply:
The 10000 was there before for the ARA trace. I ignore if there was any rationale behind this number, but I chose to preserve it just in case.

Reply:
I've checked that using a single trace for this calculation makes my laptop suffer.